Signal/src/call/CallService.swift

```diff
@@ -831,205 +878,22 @@ fileprivate let timeoutSeconds = 60
         terminateCall()
     }
 
+    /**
+     * Clean up any existing call state and get ready to receive a new call.
+     */
     private func terminateCall() {
         assertOnSignalingQueue()
-
-//        lockManager.updatePhoneState(LockManager.PhoneState.PROCESSING);
-//        NotificationBarManager.setCallEnded(this);
-//
-//        incomingRinger.stop();
-//        outgoingRinger.stop();
-//        outgoingRinger.playDisconnected();
-//
-//        if (peerConnection != null) {
-//            peerConnection.dispose();
-//            peerConnection = null;
-//        }
-//
-//        if (eglBase != null && localRenderer != null && remoteRenderer != null) {
-//            localRenderer.release();
-//            remoteRenderer.release();
-//            eglBase.release();
-//        }
-//
-//        shutdownAudio();
-//
-//        this.callState         = CallState.STATE_IDLE;
-//        this.recipient         = null;
-//        this.callId            = null;
-//        this.audioEnabled      = false;
-//        this.videoEnabled      = false;
-//        this.pendingIceUpdates = null;
-//        lockManager.updatePhoneState(LockManager.PhoneState.IDLE);
+        Logger.debug("\(TAG) in \(#function)")
 
         peerConnectionClient?.terminate()
+
         peerConnectionClient = nil
```

Comment:
Can we do something like this too:

```
peerConnectionClient .delegate = nil
```

This kind of defensive gesture would prevent CallService receiving any events from old connections.